### PR TITLE
Make sure the `/events` page is not hardcoded

### PR DESCRIPTION
**Pull request by mitogh**
🎟  https://central.tri.be/issues/98463
The URL might change due a different slug based on the options page,
make sure this URL is not hardcoded.

### CODE REVIEW

File: src/Tribe/Rewrite.php [hunk 1]
```diff
@@ -255,11 +255,12 @@ public function get_bases( $method = 'regex' ) {
 			'tag' => array( 'tag', $tec->tag_slug ),
 			'tax' => array( 'category', $tec->category_slug ),
 			'page'     => array( 'page', esc_html_x( 'page', 'The "/page/" URL string component.', 'the-events-calendar' ) ),
-			'single' => array( 'event', $tec->rewriteSlugSingular ),
-			'archive' => array( 'events', $tec->rewriteSlug ),
+			'single' => array( Tribe__Settings_Manager::get_option( 'singleEventSlug', 'event' ), $tec->rewriteSlugSingular ),
```
Comment (barryhughes) [inline src/Tribe/Rewrite.php]: This improves things but as I think about it, shouldn't we do something like:

```php
array_filter( array(
	'event',
	Tribe__Settings_Manager::get_option( 'singleEventSlug', 'event' ),
	$tec->rewriteSlugSingular
) )
```

Looking at the comment above this block, we always want to support the original/English slug - but right now we potentially will miss that element.

Comment (bordoni) [reply src/Tribe/Rewrite.php]: Uhnn I was thinking maybe the intention was to allow the users to completely remove `events/event` using a different option on the DB.

File: readme.txt [hunk 2]
```diff
@@ -309,8 +309,7 @@ The plugin is made with love by [Modern Tribe Inc](http://m.tri.be/2s).
 == Changelog ==
 
 = [4.6.10.1] 2018-01-25 =
-
-
+* Fix - Make sure rewrite rule for `/events` is not hardcoded and is based on dynamic option field [98463]
```
Comment (barryhughes) [inline readme.txt]: Especially with a hotfix, it's a nice touch to acknowledge folks who highlighted the problem. Maybe add: _(thanks to elecevan, nickfetner and everyone else who reported this problem)_

File: src/Tribe/Rewrite.php (hunk 1 above)
Comment (mitogh) [reply src/Tribe/Rewrite.php]: This one should be only for the single view? as if we do it for `archive`  we are still going to have the same issue if we leave `events` @barryhughes

File: readme.txt (hunk 2 above)
Comment (mitogh) [reply readme.txt]: For sure, will update this one.

File: src/Tribe/Rewrite.php (hunk 1 above)
Comment (barryhughes) [reply src/Tribe/Rewrite.php]: Hmm, don't think it is but if I'm wrong let's also update the comment above this section.

Seems though that we'd be creating a headache in scenarios where URLs using the original slug are promulgated via sharing links etc.

Comment (barryhughes) [reply src/Tribe/Rewrite.php]: > This one should be only for the single view?

Comment (mitogh) [reply src/Tribe/Rewrite.php]: if we leave `events` hardcoded is going to be present anyways  so we are going to have the `events` url taken by the plugin so I think we should update the documentation around this, plus is the default value if the option value is not changed if will be different only if the user change is intentionality. 

cc @bordoni @barryhughes

Comment (barryhughes) [reply src/Tribe/Rewrite.php]: I think the sane default is for us to support:

Comment (barryhughes) [reply src/Tribe/Rewrite.php]: ... filtering out any duplicates

Comment (barryhughes) [reply src/Tribe/Rewrite.php]: Actually, I see what you're saying. Let's keep this as is.

**Pull request merged**